src/parse-date.js

```diff
@@ -3,6 +3,26 @@
 var expirationYear = require('./expiration-year');
 var isArray = require('./lib/is-array');
 
+function getLen(value) {
+  if (value[0] === '0') {
+    return 2;
+  }
+
+  if (value.length > 5) {
+    return 2;
+  }
+
+  if (value[value.length - 1] === '0') {
```

Comment:
Can you explain why you've added this logic?

Reply:
this logic was laid down earlier
`'120' => {month: '1', year: '20'}`

Reply:
What does the last value being `0` have anything to do with this?

With the implementation on master:

```js
var cv = require('card-validator');

cv.expirationDate('120'); // {month: '1', year: '20', isValid: true, isPotentiallyValid: true}
cv.expirationDate('121'); // {month: '1', year: '21', isValid: true, isPotentiallyValid: true}
cv.expirationDate('122'); // {month: '1', year: '22', isValid: true, isPotentiallyValid: true}
cv.expirationDate('123'); // {month: '1', year: '23', isValid: true, isPotentiallyValid: true}
// ...
cv.expirationDate('138'); // {month: '1', year: '38', isValid: true, isPotentiallyValid: true}
```

It's not until we get to `cv.expirationDate('139');` that we get to `{month: null, year: null, isValid: false, isPotentiallyValid: false}`, and that's only because that's the point where it exceeds the default `maxElapsedYear` value of 19.

Reply:
I draw your attention to the fact that this logic was laid earlier in the tests that I closed by adding this rule. You're right. I shouldn't have closed them, I should have corrected them as an invalid value. Let me know if you agree with me and I will remove this rule and change the tests